drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,392 @@
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		icm42605_convert_gyro(val, drv_data->gyro_y,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		icm42605_convert_gyro(val, drv_data->gyro_z,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	default: /* chan == SENSOR_CHAN_DIE_TEMP */
+		icm42605_convert_temp(val, drv_data->temp);
+	}
+
+	return 0;
+}
+
+static int icm42605_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	int result = 0;
+	uint16_t fifo_count = 0;
+	static uint8_t data[HARDWARE_FIFO_SIZE] = { 0, };
+	struct icm42605_data *drv_data = dev->data;
+
+	if (drv_data->tap_en) {
+		result = inv_spi_read(REG_INT_STATUS3, data, 1);
+
+		if (data[0] & BIT_INT_STATUS_TAP_DET) {
+			LOG_DBG("Tap detected!!!");
+			result = inv_spi_read(REG_APEX_DATA4, data, 1);
+			if (data[0] & 0x08) {
+				LOG_DBG("Single Tap");
+			} else if (data[0] & 0x10) {
+				LOG_DBG("Double Tap");
+			} else {
+				LOG_DBG("Not supported");
+			}
+		}
+	}
+
+	/* Read INT_STATUS (0x45) and FIFO_COUNTH(0x46), FIFO_COUNTL(0x47) */
+	result = inv_spi_read(REG_INT_STATUS, data, 3);
+
+	if (data[0] & BIT_INT_STATUS_DRDY) {
+		fifo_count = ((data[1] & 0xff) << 8) + (data[2] & 0xff);
+		result = inv_spi_read(REG_FIFO_DATA, data, fifo_count);
+
+		/* FIFO Data structure
+		 * Packet 1 : FIFO Header(1), AccelX(2), AccelY(2), AccelZ(2), Temperature(1)
+		 * Packet 2 : FIFO Header(1), GyroX(2), GyroY(2), GyroZ(2), Temperature(1)
+		 * Packet 3 : FIFO Header(1), AccelX(2), AccelY(2), AccelZ(2), GyroX(2), GyroY(2), GyroZ(2), Temperature(1)
+		 */
+
+		if (data[0] & BIT_FIFO_HEAD_ACCEL) {
+			/* Check empty values */
+			if (!(data[1] == 0x80 && data[2] == 0x00)) {
+				drv_data->accel_x = ((data[1] & 0xff) << 8) + (data[2] & 0xff);
+				drv_data->accel_y = ((data[3] & 0xff) << 8) + (data[4] & 0xff);
+				drv_data->accel_z = ((data[5] & 0xff) << 8) + (data[6] & 0xff);
+			}
+			if (!(data[0] & BIT_FIFO_HEAD_GYRO)) {
+				drv_data->temp = (int16_t)data[7];
+			} else {
+				if (!(data[7] == 0x80 && data[8] == 0x00)) {
+					drv_data->gyro_x = ((data[7] & 0xff) << 8) + (data[8] & 0xff);
+					drv_data->gyro_y = ((data[9] & 0xff) << 8) + (data[10] & 0xff);
+					drv_data->gyro_z = ((data[11] & 0xff) << 8) + (data[12] & 0xff);
+				}
+				drv_data->temp = (int16_t)data[13];
+			}
+		} else {
+			if (data[0] & BIT_FIFO_HEAD_GYRO) {
+				if (!(data[1] == 0x80 && data[2] == 0x00)) {
+					drv_data->gyro_x = ((data[1] & 0xff) << 8) + (data[2] & 0xff);
+					drv_data->gyro_y = ((data[3] & 0xff) << 8) + (data[4] & 0xff);
+					drv_data->gyro_z = ((data[5] & 0xff) << 8) + (data[6] & 0xff);
+				}
+				drv_data->temp = (int16_t)data[7];
+			}
+		}
+	}
+
+	return 0;
+}
+
+static int icm42605_attr_set(struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     const struct sensor_value *val)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	if (val == NULL) {
```

Comment:
An assertion would be better, as having it NULL is a programming error.

Reply:
Good point. I will fix and will test.

Reply:
Fixed